BossMod/Modules/Endwalker/TreasureHunt/LyssaChrysine.cs

```diff
@@ -0,0 +1,172 @@
+    HeavySmash2 = 32317, // BossAdd->location, 3,0s cast, range 6 circle
+    FrigidStone = 32308, // Boss->self, 2,5s cast, single-target, activates helpers
+    FrigidStone2 = 32309, // BossHelper->location, 3,0s cast, range 5 circle
+};
+
+class HeavySmash2 : Components.LocationTargetedAOEs
+    {
+        public HeavySmash2() : base(ActionID.MakeSpell(AID.HeavySmash2), 6) { }
+    }
+class FrigidStone2 : Components.LocationTargetedAOEs
+    {
+        public FrigidStone2() : base(ActionID.MakeSpell(AID.FrigidStone2), 5) { }
+    }
+class OutInAOE : Components.ConcentricAOEs
+    {
+        private static AOEShape[] _shapes = {new AOEShapeCircle(10), new AOEShapeDonut(10, 20)};
+
+        public OutInAOE() : base(_shapes) { }
+
+        public override void OnCastStarted(BossModule module, Actor caster, ActorCastInfo spell)
+        {
+            if ((AID)spell.Action.ID == AID.FrigidNeedle)
+                AddSequence(module.Bounds.Center, spell.FinishAt);
+        }
+
+        public override void OnEventCast(BossModule module, Actor caster, ActorCastEvent spell)
+        {
+            var order = (AID)spell.Action.ID switch
+            {
+                AID.FrigidNeedle2 => 0,
+                AID.CircleOfIce2 => 1,
+                _ => -1
+            };
+            if (!AdvanceSequence(order, module.Bounds.Center, module.WorldState.CurrentTime.AddSeconds(4f)))
+                module.ReportError(this, $"Unexpected ring {order}");
+        }
+    }
+class InOutAOE : Components.ConcentricAOEs
+    {
+        private static AOEShape[] _shapes = {new AOEShapeDonut(10, 20), new AOEShapeCircle(10)};
+
+        public InOutAOE() : base(_shapes) { }
+
+        public override void OnCastStarted(BossModule module, Actor caster, ActorCastInfo spell)
+        {
+            if ((AID)spell.Action.ID == AID.CircleOfIce)
+                AddSequence(module.Bounds.Center, spell.FinishAt);
+        }
+
+        public override void OnEventCast(BossModule module, Actor caster, ActorCastEvent spell)
+        {
+            var order = (AID)spell.Action.ID switch
+            {
+                AID.CircleOfIce2 => 0,
+                AID.FrigidNeedle2 => 1,
+                _ => -1
+            };
+            if (!AdvanceSequence(order, module.Bounds.Center, module.WorldState.CurrentTime.AddSeconds(4f)))
+                module.ReportError(this, $"Unexpected ring {order}");
+        }
+    }
+class PillarPierce : Components.SelfTargetedAOEs
+    {
+        public PillarPierce() : base(ActionID.MakeSpell(AID.PillarPierce), new AOEShapeRect(80,2)) { }
+    }
+class SkullDasher : Components.SingleTargetCast
+    {
+        public SkullDasher() : base(ActionID.MakeSpell(AID.SkullDasher)) { }
+    }
+class HeavySmash : Components.StackWithCastTargets
+    {
+        public HeavySmash() : base(ActionID.MakeSpell(AID.HeavySmash), 6) { }
+    }
+class IcePillarSpawn : Components.GenericAOEs
+    {     
+        private float activePillar;       
```

Comment:
Should be an `Actor? _activePillar`? and then all functions should check if non-null and use it directly, rather than looking up in actor list.